sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionAsyncContainer.java

```diff
@@ -1049,72 +1125,88 @@ private <T> Function<CosmosPagedFluxOptions, Flux<FeedResponse<T>>> transformQue
         Transformer<T> transformer,
         SqlQuerySpec sqlQuerySpec,
         CosmosQueryRequestOptions queryRequestOptions,
-        CosmosPagedFluxOptions pagedFluxOptions) {
+        CosmosPagedFluxOptions pagedFluxOptions,
+        CosmosItemSerializer effectiveSerializer) {
 
-        CosmosQueryRequestOptions finalOptions = setRequestHeaders(queryRequestOptions);
+        CosmosQueryRequestOptions finalOptions = setRequestHeaders(cosmosQueryRequestOptionsAccessor
+            .clone(queryRequestOptions)
+            .setCustomSerializer(CosmosItemSerializer.DEFAULT_SERIALIZER));
 
         return transformer.transform(
             cosmosAsyncContainerAccessor.queryItemsInternalFunc(
                 this.container,
                 sqlQuerySpec,
                 finalOptions,
-                JsonNode.class)
+                JsonNode.class),
+            effectiveSerializer
         );
     }
 
     private <T> Function<CosmosPagedFluxOptions, Flux<FeedResponse<T>>> transformQueryChangeFeedInternal(
         Transformer<T> transformer,
         CosmosChangeFeedRequestOptions changeFeedRequestOptions,
-        CosmosPagedFluxOptions pagedFluxOptions) {
+        CosmosPagedFluxOptions pagedFluxOptions,
+        CosmosItemSerializer effectiveSerializer) {
 
-        CosmosChangeFeedRequestOptions finalOptions = setRequestHeaders(changeFeedRequestOptions);;
+        CosmosChangeFeedRequestOptions finalOptions = setRequestHeaders(
+            cosmosChangeFeedRequestOptionsAccessor.clone(changeFeedRequestOptions)
+                .setCustomSerializer(CosmosItemSerializer.DEFAULT_SERIALIZER));
         getEffectiveCosmosChangeFeedRequestOptions(pagedFluxOptions, finalOptions);
 
         return transformer.transform(
             cosmosAsyncContainerAccessor
                 .queryChangeFeedInternalFunc(
                     this.container,
                     finalOptions,
-                    JsonNode.class)
+                    JsonNode.class),
+            effectiveSerializer
         );
     }
 
     private <T> Function<CosmosPagedFluxOptions, Flux<FeedResponse<T>>> transformQueryItemsWithMonoSqlQuerySpec(
         Transformer<T> transformer,
         Mono<SqlQuerySpec> sqlQuerySpecMono,
         CosmosQueryRequestOptions options,
-        CosmosPagedFluxOptions pagedFluxOptions) {
+        CosmosPagedFluxOptions pagedFluxOptions,
```

Comment:
Likewise - `pagedFluxOptions` can be removed.

Reply:
Fixed